lm_eval/tasks/periodic_table.py

```diff
@@ -0,0 +1,70 @@
+"""
+PeriodicTable is a custom evaluation task for them ChemNLP project.
+
+This Q/A task was programatically created from the periodic table.
+The task is about the contents and structure of the periodic table only.
+
+"""
+from lm_eval.base import MultipleChoiceTask
+
+
+# TODO: How will we cite our new tasks?
+
+
+class PeriodicTable(MultipleChoiceTask):
+    VERSION = 0
+    DATASET_PATH = "OpenBioML/PeriodicTable"
+    DATASET_NAME = None
+
+    def has_training_docs(self):
+        return False
+
+    def has_validation_docs(self):
+        return True
+
+    def has_test_docs(self):
+        return True
+
+    def validation_docs(self):
+        return map(self._process_doc, self.dataset["validation"])
+
+    def test_docs(self):
+        if self.has_test_docs():
+            return map(self._process_doc, self.dataset["test"])
+
+    def _process_doc(self, doc):
+        def format_example(doc, keys):
+            """
+            Question: <prompt>
+            Choices:
+            A. <choice1>
+            B. <choice2>
+            C. <choice3>
+            D. <choice4>
+            Answer:
+            """
+            prompt = "Question: " + doc["question"] + "\nChoices:\n"
+            prompt += "".join(
+                [f"{key}. {choice}\n" for key, choice in zip(keys, doc["choices"])]
+            )
+            prompt += "Answer:"
+            return prompt
+
+        keys = ["A", "B", "C", "D"]
```

Comment:
I think when creating the validation / test documents with `map(...)` we will instantiate this variable as a new value every time but it's actual a constant.

You could add it as a class attribute constant and then reference `self.KEYS`?

Reply:
same as above, this is from the template so I will leave it as is for now so it matches the other multichoice tasks e.g. Hendrycks. But maybe in future if we section out to make our own code base which is not part of lm-eval we can make these improvements?